recipes/spirv-tools/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+    description = "Create and optimize SPIRV shaders"
+    topics = ("conan", "spirv", "spirv-v", "vulkan", "opengl", "opencl", "hlsl", "khronos")
+    url = "https://github.com/conan-io/conan-center-index"
+    short_paths = True
+    settings = "os", "compiler", "arch", "build_type"
+    exports_sources = ["CMakeLists.txt"]
+    license = "Apache-2.0"
+    generators = "cmake"
+    _cmake = None
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True
+    }
+
+    def requirements(self):
+        self.requires.add("spirv-headers/1.5.1")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "SPIRV-Tools-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        cmake = CMake(self)
+
+        if self.options.shared and self.settings.compiler == "Visual Studio":
+            cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+
+        # Required by the project's CMakeLists.txt
+        cmake.definitions["SPIRV-Headers_SOURCE_DIR"] = self.deps_cpp_info["spirv-headers"].rootpath
+
+        # There are some switch( ) statements that are causing errors
+        # need to turn this off
+        cmake.definitions["SPIRV_WERROR"] = False
+
+        cmake.definitions["SKIP_SPIRV_TOOLS_INSTALL"] = False
+        cmake.definitions["SPIRV_LOG_DEBUG"] = False
+        cmake.definitions["SPIRV_SKIP_TESTS"] = True
+        cmake.definitions["SPIRV_CHECK_CONTEXT"] = False
+        cmake.definitions["SPIRV_BUILD_FUZZER"] = False
+
+        cmake.configure(build_folder=self._build_subfolder)
+        self._cmake = cmake
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _patch_sources(self):
+        # CMAKE_POSITION_INDEPENDENT_CODE was set ON for the entire
+        # project in the lists file.
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "set(CMAKE_POSITION_INDEPENDENT_CODE ON)", "")
+
+    def package(self):
+        self.copy(pattern="LICENSE*", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        # Error KB-H020, complaining that .pc files are found
+        tools.rmdir(os.path.join(self.package_folder, "lib/pkgconfig"))
+
+        # Error KB-H019, complaining that .pc files are found
+        tools.rmdir(os.path.join(self.package_folder, "lib/cmake"))
```

Comment:
```suggestion
        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```

Also I think that comments related to hooks are unecessary, it's very common to remove those folders in package method.